DEVELOPING.md

```diff
@@ -0,0 +1,20 @@
+# For Developers
+
+We strongly encourage you to review the project's scope described in the `README.md` file before working on new features. For large changes, consider writing a design document using [this template](https://docs.google.com/document/d/1U6GlAaPF3tl1wvBNXebW1mD3oHZxV5h9TH-bR0p8Uz8/edit?usp=sharing).
```

Comment:
Document is not public (pointed out by @filipesilva in #348)

Reply:
Thanks for pointing this out! Will sync today.